griptape/mixins/runnable_mixin.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Generic, Optional, TypeVar, cast
+
+from attrs import define, field
+
+# Generics magic that allows us to reference the type of the class that is implementing the mixin
+T = TypeVar("T", bound="RunnableMixin")
+
+
+@define()
+class RunnableMixin(ABC, Generic[T]):
```

Comment:
Maybe worth a comment that when used as a parent class, `T` must be the the child class.

For example, this will work:
```python
class Foo(RunnableMixin["Foo"]):
   ...
```

But this will result in weird errors:
```python
class Foo(RunnableMixin[Bar]):
   ...
```

Reply:
Added docstring to mixin.